lang/cs/Org.Apache.REEF.IMRU/OnREEF/Driver/IMRUDriver.cs

```diff
@@ -593,9 +599,9 @@ public void OnNext(IFailedContext failedContext)
         {
             lock (_lock)
             {
-                if (_taskManager.AreAllTasksCompleted())
+                if (_systemState.CurrentState.Equals(SystemState.TasksCompleted))
```

Comment:
Why not keep this an expressive method call? If we ever change the meaning of the state variable, we will have to remember to touch this line, otherwise.,